### ISSUE TITLE
Screen flashes when entering text in dialog screen on certain IOS models.

### ISSUE THREAD

**Issue by hatano0x06** [flutter/flutter#155502]
### Steps to reproduce

This is a report from a user of my app.
This only happens on certain devices of certain models, and I don't know if it can be reproduced.
I have not been able to reproduce it.
I have confirmed this on iPhone 16,1 and iPad 8,11.

All I know is that when I enter text in the textformfield on the dialog page, the screen blinks.

I don't know if I can reproduce it, but I will draw the steps to reproduce it.

1. input text about 8 lines

### Expected results

I can type normally.
No abnormality on the screen

### Actual results

The screen flickers.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:modal_bottom_sheet/modal_bottom_sheet.dart';

class SamplePage extends StatefulWidget {

  SamplePage(): super();

  @override
  SamplePageState createState() => SamplePageState();
}

class SamplePageState extends State<SamplePage> {



  @override
  Widget build(BuildContext context) {

    return CupertinoPageScaffold(
      navigationBar : CupertinoNavigationBar(
        transitionBetweenRoutes   : false,
        middle          : Text("Sample Dialog"),
      ),
      child: Container(
        padding: EdgeInsets.only(top: 250,),
        child: CupertinoButton(child: Text("press"), onPressed: (){
          showBarModalBottomSheet(
            context: context,
            expand: true,
            backgroundColor : Colors.transparent,
            topControl  : SizedBox(height: kToolbarHeight*3/4,),
            duration    : Duration(milliseconds: 300),
            builder         : (context) => SampleDialog(),
          );      

        },),
      ),
    );
  }

}

class SampleDialog extends StatefulWidget {
  const SampleDialog() : super();

  @override
  _SampleDialogState createState() => _SampleDialogState();
}

class _SampleDialogState extends State<SampleDialog> with WidgetsBindingObserver {
  final TextEditingController _textEditingController = TextEditingController();
  final FocusNode _focusNode = FocusNode();
  final UndoHistoryController _textUndoController = UndoHistoryController();

  @override
  void initState(){
    super.initState(); 
    Future.delayed(Duration(milliseconds: 100)).then((_){
      if (!this.mounted) return;
      FocusScope.of(context).requestFocus(_focusNode);
    });
  }
  
  @override
  void dispose() {
    WidgetsBinding.instance.removeObserver(this);

    _textEditingController.dispose();
    _focusNode.dispose();
    _textUndoController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {

    return CupertinoPageScaffold(
      child: SingleChildScrollView(
        child: Column(children: [
          SizedBox(height: 100,),
          Container(
            height: 500,
            color: Colors.blue,
            padding: EdgeInsets.all(20),
            child : CupertinoTextField(
              keyboardType : TextInputType.multiline,
              maxLines     : null,
              placeholder: "input here",
            ),
          ),

        ],),
      ),
      navigationBar : CupertinoNavigationBar(
        transitionBetweenRoutes   : false,
        middle          : Text("Sample Dialog"),
      )
    );
  }

}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/feb24efe-6b7e-45c4-8e9c-2dd715f68be5

https://github.com/user-attachments/assets/80ec14ca-8b8f-40c2-96f3-d3a2d3adc5de

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
MacBook-Air:mangaplotter hata$ flutter doctor -v
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.5 23F79 darwin-arm64, locale ja-JP)
  • Flutter version 3.24.3 on channel stable at /Users/hata/Desktop/develop/flutter
  • Upstream repository https://github.com/flutter/flutter.git
  • Framework revision 2663184aa7 (12 days ago), 2024-09-11 16:27:48 -0500
  • Engine revision 36335019a8
  • Dart version 3.5.3
  • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
  • Android SDK at /Users/hata/Library/Android/sdk
  • Platform android-34, build-tools 31.0.0
  • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
  • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
  • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
  • Xcode at /Applications/Xcode.app/Contents/Developer
  • Build 15F31d
  • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
  • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
  • Android Studio at /Applications/Android Studio.app/Contents
  • Flutter plugin can be installed from:
   :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
  • Dart plugin can be installed from:
   :hammer: https://plugins.jetbrains.com/plugin/6351-dart
  • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] Connected device (4 available)
  • hataのiPad (2) (mobile)     • 00008101-000149491E8B001E • ios      • iOS 17.6.1 21G93
  • macOS (desktop)         • macos           • darwin-arm64  • macOS 14.5 23F79 darwin-arm64
  • Mac Designed for iPad (desktop) • mac-designed-for-ipad   • darwin     • macOS 14.5 23F79 darwin-arm64
  • Chrome (web)          • chrome          • web-javascript • Google Chrome 128.0.6613.139

[✓] Network resources
  • All expected network resources are available.

• No issues found!

```

</details>

**Comment by hatano0x06** [comment]
This could be a problem with my implementation.
However, this is the first time that the entire screen is flickering, not just the problem area, and I don't know how to deal with it!

I would be glad to know in what cases this problem occurs!

**Comment by huycozy** [comment]
@hatano0x06 [modal_bottom_sheet](https://pub.dev/packages/modal_bottom_sheet) is a 3rd party package. Can you try using [showModalBottomSheet](https://api.flutter.dev/flutter/material/showModalBottomSheet.html) from framework to see if the issue persists or not?

Also, your video demo doesn't seem to match with sample code. Do you see the issue on a minimal sample code (without complex integration with unrelated widgets or 3rd party plugins)? I'm not sure if you have a macOS machine (since the flutter doctor is showing on Windows machine), so if you have one, you can try on iOS emulator to see if can reproduce issue.

**Comment by hatano0x06** [comment]
Thanks for pointing that out, I'll make it showModalBottomSheet and release it.

The reason why the video is different from the minimal code is because I have only discovered it through reports from my app users, and the video was provided by an app user!
Of course, the app is implemented in a huge code, so I can't release everything here!
I am providing the minimal code with only the necessary parts extracted.

As for the doctor, we have updated it to the mac version!



Translated with DeepL.com (free version)
